### PR TITLE
Fix Warning in ArgumentRenderer #1667

### PR DESCRIPTION
**Pull request by mitsuruog**
Resolved

- Warning in ArgumentRenderer #1667

I checked the behavior of the component of the original issue after I changed the `name` to optional and found that the following error occurred.

```
DoctrineError {name: "DoctrineError", message: "Unknown type undefined"}
```

What I have found in my research is that in the above cases, `type.type` in the `returns` is lost in `getProps` in this case.
`returns` should have `NameExpression` in `type.type` in my opinion.

```js
// I saw this shape.
{ 
  description: 'return a Boolean Value', 
  type: { name: 'boolean' } 
}

// but it should be this shape.
{
  title: 'returns',
  description: 'return a Boolean Value',
  type: { type: 'NameExpression', name: 'boolean' }
}
```

### CODE REVIEW

File: src/loaders/utils/getProps.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export default function getProps(doc: DocumentationObject, filepath?: string): R
 			allTags as TagProps,
 			JS_DOC_METHOD_RETURN_TAG_SYNONYMS
 		) as TagParamObject[];
-		const returns = method.returns || returnTags[0];
+		const returns = (method.returns || returnTags[0]) && { ...method.returns, ...returnTags[0] };
```
Comment (sapegin) [inline src/loaders/utils/getProps.ts:97]: Looks like this isn't different from the previous version, unless I'm missing something (in this case we should make it more clear).

We're returning `method.returns` if it's truthy, or `returnTags[0]`.

**Review by sapegin — CHANGES_REQUESTED** [review]
Thanks for the pull request!

File: src/loaders/utils/getProps.ts (hunk 1 above)
Comment (mitsuruog) [reply src/loaders/utils/getProps.ts:97]: hmm... we need to make it more clear. in my case, `method.returns` have this object:

```js
{ 
  description: 'return a Boolean Value', 
  type: { name: 'boolean' } 
}
```

and then `returnTags[0]` have this object:

```js
{
  title: 'returns',
  description: 'return a Boolean Value',
  type: { type: 'NameExpression', name: 'boolean' }
}
```

`method.returns` will have occurred `DoctrineError` when it does not have `{ type: 'NameExpression' }`.
so I figured that merging both results would result in the intended object shape. 
But I'm also sure I don't understand enough about this logic.

What should the correct object shape of `results` be?

Comment (sapegin) [reply src/loaders/utils/getProps.ts:97]: Now I'm even more confused. So now we're overwriting anything in `method.returns` with `returnTags[0]` but previously `method.returns`  had higher priority.

If `method.returns`  is missing the type, why not add it there? Something like this:

```
const getReturns = ({method, tags}) => {
                 if (method.returns) {
                       return {
                           ...method.returns,
                           type: { type: 'NameExpression', ...method.returns.type } 
                       }
                 }

		const returnTags = getMergedTag(
			tags as TagProps,
			JS_DOC_METHOD_RETURN_TAG_SYNONYMS
		) as TagParamObject[];

                 return returnTags[0];
}
```

Comment (mitsuruog) [reply src/loaders/utils/getProps.ts:97]: > previously `method.returns` had higher priority.

I got it. I’ll add a change on the `method.returns` side.

**Comment by mitsuruog** [comment]
@sapegin please review me.

**Review by sapegin — APPROVED** [review]
🦄

**Pull request merged**